benchmarking/benchmark.py

```diff
@@ -0,0 +1,380 @@
+    else:
+        fp_function = lambda x: generator.GetFingerprint(MolFromSmiles(x))
+
+    # testing for different sizes of input datasets
+    result = []
+    for data_fraction in np.linspace(0, 1, N_SPLITS + 1)[1:]:
+        n = int(n_molecules * data_fraction)
+        subset = X[:n]
+        times = [None for _ in range(N_REPEATS)]
+        # testing several times to get average computation time
+        for i in range(N_REPEATS):
+            start = time()
+            if sparse:
+                X_transformed = csr_array([fp_function(x) for x in subset])
+            else:
+                X_transformed = np.array([fp_function(x) for x in subset])
+            end = time()
+            times[i] = end - start
+        result.append(np.mean(times))
+    return np.array(result)
+
+
+def get_times_rdkit(
+    X: pd.DataFrame, func: Callable, sparse: bool = False, **kwargs
+):
+    n_molecules = X.shape[0]
+    # testing for different sizes of input datasets
+    result = []
+    for data_fraction in np.linspace(0, 1, N_SPLITS + 1)[1:]:
+        n = int(n_molecules * data_fraction)
+        subset = X[:n]
+        times = [None for _ in range(N_REPEATS)]
+        # testing several times to get average computation time
+        for i in range(N_REPEATS):
+            start = time()
+            if sparse:
+                X_transformed = csr_array(
+                    [func(MolFromSmiles(x), **kwargs) for x in subset]
+                )
+            else:
+                X_transformed = np.array(
+                    [func(MolFromSmiles(x), **kwargs) for x in subset]
+                )
+            end = time()
+            times[i] = end - start
+        result.append(np.mean(times))
+    return np.array(result)
+
+
+def plot_results(
+    n_molecules: int,
+    y_emf: List,
+    y_rdkit: List,
+    title: str = "",
+    sparse: bool = None,
+    count: bool = None,
+    save: bool = True,
+):
+    if sparse is not None:
+        if sparse:
+            title += " sparse"
+
+    if count is not None:
+        if count:
+            title += " count"
+        else:
+            title += " bit"
+
+    X = n_molecules * np.linspace(0, 1, N_SPLITS + 1)[1:]
+
+    plt.rcParams["font.size"] = 20
+    fig = plt.figure(figsize=(15, 10))
+    ax1 = fig.add_subplot()
+    ax1.set_title(title)
+
+    ax1.plot(X, y_emf[0], label="emf time - 1 job")
+    ax1.plot(X, y_emf[1], label="emf time - 2 job")
+    ax1.plot(X, y_emf[2], label="emf time - 4 job")
+    ax1.plot(X, y_emf[3], label="emf time - all jobs")
+    ax1.plot(X, y_rdkit, label="rdkit time")
+
+    ax1.set_ylabel("Time of computiation")
+    ax1.set_xlabel("Number of finberprints")
+
+    ax1.set_xlim(n_molecules * 0.1, n_molecules * 1.1)
+    ax1.set_ylim(bottom=0)
+
+    plt.legend(loc="upper left")
+    if save:
+        plt.savefig(title.replace(" ", "_") + ".png")
+    else:
+        plt.show()
+
+
+if __name__ == "__main__":
+    # GraphPropPredDataset(name=dataset_name)
```

Comment:
Should this be left here?

Reply:
The if statement or the comment?

Reply:
The commented line was supposed to get the dataset from ogb but it errors out. I can make a hotfix for it

Reply:
Hotfix done. Added a md file that explains where and how to modify the code in ogb